lib/rouge/lexers/bpf.rb

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class BPF < RegexLexer
+      title "BPF"
+      desc "BPF bytecode syntax"
+      tag 'bpf'
+
+      TYPE_KEYWORDS = %w(
+        u8 u16 u32 u64 ll
+      ).join('|')
+
+      MISC_KEYWORDS = %w(
+        be16 be32 be64 exit lock
+      ).join('|')
+
+      state :root do
+        # Calls to helpers
+        rule /(call)(\s+)(\d+)/i do |m|
+          groups Keyword, Text::Whitespace, Literal::Number::Integer
+        end
+
+        # Unconditional jumps
+        rule /(goto)(\s*)(\+\d+)?(\s*)(<?\w+>?)/i do |m|
+          groups Keyword, Text::Whitespace, Literal::Number::Integer, Text::Whitespace, Name::Label
+        end
+
+        # Conditional jumps
+        rule /(if)(\s+)(r\d+)(\s*)([s!=<>]+)(\s*)(0x\h+|[-]?\d+)(\s*)(goto)(\s*)(\+\d+)?(\s*)(<?\w+>?)/i do |m|
+          groups Keyword, Text::Whitespace, Name, Text::Whitespace, Operator, Text::Whitespace, Literal::Number, Text::Whitespace, Keyword, Text::Whitespace, Literal::Number::Integer, Text::Whitespace, Name::Label
+        end
+        rule /(if)(\s+)(r\d+)(\s*)([s!=<>]+)(\s*)(r\d+)(\s*)(goto)(\s*)(\+\d+)?(\s*)(<?\w+>?)/i do |m|
+          groups Keyword, Text::Whitespace, Name, Text::Whitespace, Operator, Text::Whitespace, Name, Text::Whitespace, Keyword, Text::Whitespace, Literal::Number::Integer, Text::Whitespace, Name::Label
+        end
+
+        # Dereferences
+        rule /(\*)(\s*)(\()(#{TYPE_KEYWORDS})(\s*)(\*)(\))/i do |m|
+          groups Operator, Text::Whitespace, Punctuation, Keyword::Type, Text::Whitespace, Operator, Punctuation
+          push :address
+        end
+
+        # Operators
+        rule /[+-\/\*&|><^s]{0,3}=/i, Operator
+
+        # Registers
+        rule /([+-]?)(r\d+)/i do |m|
+          groups Punctuation, Name
+        end
+
+        # Comments
+        rule /\/\//, Comment::Single, :linecomment
+        rule /\/\*/, Comment::Multiline, :multilinescomment
+
+        rule /#{MISC_KEYWORDS}/, Keyword
+
+        # Literals and global objects (maps) refered by name
+        rule /(0x\h+|[-]?\d+)(\s*)(ll)?/i do |m|
+          groups Literal::Number, Text::Whitespace, Keyword::Type
+        end
+        rule /(\w+)(\s*)(ll)/i do |m|
+          groups Name, Text::Whitespace, Keyword::Type
+        end
+
+        # Labels
+        rule /(\w+)(\s*)(:)/i do |m|
+          groups Name::Label, Text::Whitespace, Punctuation
+        end
+
+        rule /./m, Text
```

Comment:
Since this will match every character, it will mean there are never any errors generated for characters that aren't otherwise matches against the other rules. Is that intentional?

Reply:
Yes, it was. The lexer is likely to require updates (BPF and the Clang backend evolve rapidly), so I'd rather not have red highlighting each time that happens. I think the absence of highlighting will be enough to see if the lexer needs an update.

In general, I tend to prefer to reserve red highlighting for known syntax errors, but I understand that is a matter of opinion/convention.